src/apis/source.rs

```diff
@@ -397,7 +396,7 @@ impl SourceAPI {
 
     pub async fn get_api_key(&self, access_key_id: String) -> Result<APIKey, BackendError> {
         // Try to get the cached value
-        let cache_key = format!("{}", access_key_id);
+        let cache_key = access_key_id.to_string();
```

Comment:
Another odd `.to_string()` on a `String`:

```suggestion
        let cache_key = access_key_id.clone();
```